src/domain/template/template.entity.ts

```diff
@@ -0,0 +1,54 @@
+import { Field, ID, ObjectType } from '@nestjs/graphql';
+import {
+  BaseEntity,
+  Column,
+  Entity,
+  JoinTable,
+  ManyToMany,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+import { Ecoverse } from '../ecoverse/ecoverse.entity';
+import { User } from '../user/user.entity';
+import { ITemplate } from './template.interface';
+
+@Entity()
+@ObjectType()
+export class Template extends BaseEntity implements ITemplate {
+  @Field(() => ID)
+  @PrimaryGeneratedColumn()
+  id!: number;
+
+  @Field(() => String)
+  @Column()
+  name: string;
+
+  @Field(() => String)
+  @Column('varchar', { length: 300 })
+  description: string;
+
+  @Field(() => [User], {
+    nullable: true,
+    description:
+      'The set of user types that are available within this template',
+  })
+  @ManyToMany(
+    () => User,
+    user => user.templates,
+    { eager: true, cascade: true }
```

Comment:
Many-to-many and eager loading will be a performance disaster most likely.

Reply:
Only going to be one user near term, so will not have an impact imho...

Reply:
First lets get it working, will be very little in the template for now